private/bufpkg/bufconfig/write.go

```diff
@@ -280,6 +280,11 @@ func writeConfig(
 			return fmt.Errorf("version %q found for breaking config, does not match top level config version: %q", breakingConfigVersion, version)
 		}
 		config.Breaking = breakingConfig
+	} else {
+		// Otherwise, we set a breaking config with the given version.
+		config.Breaking = &bufbreakingconfig.Config{
+			Version: version,
+		}
```

Comment:
Are these still needed if we go with the `newModule` approach that you have below?

Reply:
My conjecture is that if we are defaulting everywhere, we should probably default on the `WriteConfig` also... but maybe that is just... too much. But yes, the code path is not _required_, it's mostly for consistency in behaviour. I can remove it for now, since it'll be one less place we'll need to deal with for the `v1` migration.